src/flatbuffers/containers/fb_index.rs

```diff
@@ -0,0 +1,45 @@
+use flatbuffers::{Follow, Vector};
+
+/// A trait to access indexed data in a flatbuffer.
+/// It has two implementations:
+/// 1. a faster &[I] for slices;
+/// 2. a slower for flatbuffers::Vector<I>, that uses Follow() internally.
+///
+/// Note: it intentionally returns values using a copy, because it's faster
+/// than by reference.
+pub(crate) trait FbIndex<I> {
+    fn len(&self) -> usize;
+    fn get(&self, index: usize) -> I;
```

Comment:
None of the implementations build in any length checks, so overflowed accesses may panic. Let's make sure that is documented in the trait.

Reply:
Thanks. I've added comments.
Just to clarify: `get()` does range check every time (that's not always good for perf), but panics if it fails.